hbase-server/src/main/java/org/apache/hadoop/hbase/master/ServerManager.java

```diff
@@ -921,8 +921,10 @@ public boolean isClusterShutdown() {
   public void startChore() {
     Configuration c = master.getConfiguration();
     if (persistFlushedSequenceId) {
-      // when reach here, RegionStates should loaded, firstly, we call remove deleted regions
-      removeDeletedRegionFromLoadedFlushedSequenceIds();
+      new Thread(() -> {
```

Comment:
We will both touch flushedSequenceIdByRegion and storeFlushedSequenceIdsByRegion in persistRegionLastFlushedSequenceIds and removeDeletedRegionFromLoadedFlushedSequenceIds? Possible data race if we run removeDeletedRegionFromLoadedFlushedSequenceIds in background?

Reply:
1)flushedSequenceIdByRegion and storeFlushedSequenceIdsByRegion are ConcurrentSkipListMap
2)flushedSequenceIdByRegion and storeFlushedSequenceIdsByRegion are used only when splitting logs and updating flushSeqId, and the rs only get regionSeqId once during split log...
i think it is ok, lik memstore put and scan at the same time

Reply:
I think the problem here is that, we are in an initialization process when calling removeDeletedRegionFromLoadedFlushedSequenceIds, so it is safe to check regionStates, and then delete regions in flushedSequenceIdByRegion and storeFlushedSequenceIdsByRegion. If we do this in a background thread, after initializing, HMaster could change regionStates at the same time, is it possible that we think the region is not exists when checking regionState but then it is added back?
Maybe it is impossible as a new region should never be flushed before adding into the regionState? Anyway, let's add more comments here to tell later developers that executing it in background is safe.

Reply:
yes, Before removeDeletedRegionFromLoadedFlushedSequenceIds, the all regions have been loaded into regionstates.
and it is impossible that the region is not exists when checking regionState but then it is added back
ok, will add more comments...